backend/src/main/java/com/example/backend/filter/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,74 @@
+package com.example.backend.filter;
+
+import java.io.IOException;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import com.example.backend.service.JwtService;
+
+import io.jsonwebtoken.Claims;
+
+@Component
+@RequiredArgsConstructor
+// validates the access token before any request that needs authorization
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private final JwtService jwtService;
+    private final UserDetailsService userDetailsService;
+
+    @Override
+    protected void doFilterInternal(
+            @NonNull HttpServletRequest request,
+            @NonNull HttpServletResponse response,
+            @NonNull FilterChain filterChain) throws ServletException, IOException {
+        final String authHeader = request.getHeader("Authorization");
+
+        // authentication header should be of the form "Bearer {accessToken}"
+        if (authHeader == null || !authHeader.startsWith("Bearer ")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        // the rest of the string after the "Bearer" keyword
+        final String accessToken = authHeader.substring(7);
```

Comment:
Same as the last one.
You can also split the text beforehand on the first space than hardcoding substring(7)